lib/start.js

```diff
@@ -29,5 +29,69 @@
 if (process.env.ELECTRON_RUN_AS_NODE || process.env.ATOM_SHELL_INTERNAL_RUN_AS_NODE) {
   require('./cli/etcher')
 } else {
-  require('./gui/etcher')
+  const electron = require('electron')
+  const os = require('os')
+  const _ = require('lodash')
+  const Bluebird = require('bluebird')
+  const commandExists = Bluebird.promisify(require('command-exists'))
+  const childProcess = require('child_process')
+  const permissions = require('./shared/permissions')
+  const errors = require('./shared/errors')
+  const EXIT_CODES = require('./shared/exit-codes')
+
+  permissions.isElevated().then((isElevated) => {
+    if (_.includes([ 'win32', 'darwin' ], os.platform()) || isElevated) {
+      require('./gui/etcher')
+      return Bluebird.resolve()
+    }
+
+    return Bluebird.any(_.map([
+      {
+        name: 'gksudo',
+        options: [
+          '--preserve-env',
+          '--message',
+          'Please provide your password. Etcher needs special permissions to interact with raw devices.'
+        ]
+      },
+      {
+        name: 'kdesudo',
+        options: []
+      }
+    ], (command) => {
+      return commandExists(command.name).then((exists) => {
+        if (!exists) {
+          throw new Error(`Command does not exist: ${command.name}`)
+        }
+
+        return command
+      })
+    })).then((command) => {
+      return new Bluebird((resolve, reject) => {
+        const args = process.env.APPIMAGE ? [ process.env.APPIMAGE ] : process.argv
+        const appinstallSkip = !process.env.APPIMAGE ? {} : {
+          SKIP: 1
```

Comment:
We could always pass this variable (since its harmless) and reduce a bit the complexity